core/src/main/scala/sjsonnew/CalendarFormats.scala

```diff
@@ -16,14 +16,74 @@
 
 package sjsonnew
 
-import java.util.{Date, TimeZone, Calendar}
-import javax.xml.bind.DatatypeConverter
+import java.util.{ Calendar, GregorianCalendar }
+import java.time._
+import java.time.format.DateTimeFormatter
 
 trait CalendarFormats {
   self: IsoFormats =>
 
+  private val utc: ZoneId = ZoneId.of("UTC")
+
   implicit val calendarStringIso: IsoString[Calendar] = {
-    IsoString.iso[Calendar]( (c: Calendar) => DatatypeConverter.printDateTime(c),
-      (s: String) => DatatypeConverter.parseDateTime(s))
+    IsoString.iso[Calendar]( (c: Calendar) => {
+        // https://docs.oracle.com/javase/tutorial/datetime/iso/legacy.html
+        val i = c.toInstant
+        val tz = c.getTimeZone.toZoneId
+        zonedDateTimeToString(ZonedDateTime.ofInstant(i, tz))
+      },
+      (s: String) => {
+        GregorianCalendar.from(stringToZonedDateTime(s))
+      })
+  }
+
+  implicit val zonedDateTimeStringIso: IsoString[ZonedDateTime] =
+    IsoString.iso[ZonedDateTime](zonedDateTimeToString, stringToZonedDateTime)
```

Comment:
Because of the unlawful nature of this (doesn't round-trip) I think we should move it to a new trait that isn't mixed into BasicJsonProtocol and comes with a big Scaladoc warning.